pallets/phala/src/lib.rs

```diff
@@ -295,12 +302,19 @@ decl_module! {
 			// Extract quote fields
 			let raw_quote_body = parsed_report["isvEnclaveQuoteBody"].as_str().unwrap();
 			let quote_body = base64::decode(&raw_quote_body).unwrap();
-			// TODO: check the following fields
-			// let mr_enclave = &quote_body[112..143];
-			// let isv_prod_id = &quote_body[304..305];
-			// let isv_svn = &quote_body[306..307];
-			let report_data = &quote_body[368..432];
+			// Check the following fields
+			let mr_enclave = &quote_body[112..143];
+			let isv_prod_id = &quote_body[304..305];
+			let isv_svn = &quote_body[306..307];
+			let mut unknown_enclve = Vec::new();
+			unknown_enclve.extend_from_slice(&mr_enclave);
+			unknown_enclve.extend_from_slice(&isv_prod_id);
+			unknown_enclve.extend_from_slice(&isv_svn);
+			let white_list = MrEnclaveWhiteList::get();
+			let existed = white_list.iter().find(|x| x == &&unknown_enclve);
+			ensure!(!(existed == None), Error::<T>::WrongMrEnclave);
```

Comment:
I think it's better to move this to a standalone function because the function is already too long and this part is too detailed.